From 5d85a94fd0965aa26adf593b5f84948e9c502121 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Mon, 7 Oct 2019 13:22:14 +0200
Subject: [PATCH] setup.py: do not report missing dependencies for disabled
 modules

Reporting those missing dependencies is misleading as the modules would not
have been built anyway. This particularly matters in oe-core's automated
build completeness checker which relies on the report.

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>

Rebased for 3.9 [10/2020]
Signed-off-by: Alejandro Enedino Hernandez Samaniego <alejandro.hernandez@linux.microsoft.com>

---
 setup.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/setup.py b/setup.py
index ab9fb8e..8cf058c 100644
--- a/setup.py
+++ b/setup.py
@@ -495,6 +495,11 @@ class PyBuildExt(build_ext):
                 print("%-*s   %-*s   %-*s" % (longest, e, longest, f,
                                               longest, g))
 
+        # There is no need to report missing module dependencies,
+        # if the modules have been disabled in the first place.
+        sysconf_dis = sysconfig.get_config_var('MODDISABLED_NAMES').split()
+        self.missing = list(set(self.missing) - set(sysconf_dis))
+
         if self.missing:
             print()
             print("Python build finished successfully!")
